### PR TITLE
Improve loading visualization

### PR DESCRIPTION
**Pull request by timvahlbrock**
### Description

This is the Pull Request Template for the Oeffis Project

## Checklist

- [x] My code follows the style guidelines
- [x] I have documented my code
- [x] My changes generate no new warnings
- [x] I have linked the Issue
- [x] The Pull Request covers all the Checkpoints from the tagged Issue
- [x] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [x] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [x] No tests required
- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by timvahlbrock** [comment]
@Paulomart I'm aware that there is a pending state, but due to the debounce that didn't work. How should I continue?

File: frontend/src/components/LocationSearchInput.tsx [hunk 1]
```diff
@@ -25,9 +26,15 @@ export type LocationSearchInputProps = {
 };
 
 export const LocationSearchInput = (props: LocationSearchInputProps): JSX.Element => {
+  const [loading, setLoading] = useState<boolean>(false);
```
Comment (Paulomart) [inline frontend/src/components/LocationSearchInput.tsx:29]: Why introduce another state here? Could you just have used state === pending?

**Review by Paulomart — CHANGES_REQUESTED** [review]

File: frontend/src/components/LocationSearchInput.tsx (hunk 1 above)
Comment (timvahlbrock) [reply frontend/src/components/LocationSearchInput.tsx:29]: No, but don't know why. maybe because of the debounce?

Comment (Paulomart) [reply frontend/src/components/LocationSearchInput.tsx:29]: I dont get it, why dont just display the loading animation when the state is pending?

Comment (timvahlbrock) [reply frontend/src/components/LocationSearchInput.tsx:29]: I tried, but it didn't work. Feel free to try again and commit it.

Comment (timvahlbrock) [reply frontend/src/components/LocationSearchInput.tsx:29]: Also: the state is not pending when the debounce is still cooling down. Which is technically correct but I'm not sure that's what we want to show to the user.

Comment (Paulomart) [reply frontend/src/components/LocationSearchInput.tsx]: [a4bd0c6](https://github.com/Oeffis/Oeffis/pull/288/commits/a4bd0c6c7472c5c57e9929867489f03e0abbcd7b)

**Review by Paulomart — APPROVED** [review]

**Pull request merged**